course5/homework6/string.cc

```diff
@@ -0,0 +1,115 @@
+#include <memory>
+#include <cstring>
+
+#include "string.h"
+
+namespace jds {
+
+String::String()
+{
+	mLength = 0;
+}
+
+String::String(String const & that)
+{
+	mLength = that.mLength;
+	mBuffer = std::make_unique<char[]>(mLength + 1);
+	std::memcpy(this->mBuffer.get(), that.mBuffer.get(), that.mLength);
+	mBuffer.get()[mLength] = '\0';
+}
+
+String::String(String && that)
+{
+	mBuffer = std::unique_ptr<char []>(that.mBuffer.release());
```

Comment:
Tried that first last night and did not get it to compile for some reason so I thought I was doing something wrong. This worked too. Your assertion that `std::move` should work made me put it back in and apparently I did something wrong yesterday because it indeed "just works".